### PR TITLE
[REVIEW] Fix string issue for parquet reader and support `keep_index` for `scatter_to_tables`

### PR DESCRIPTION
**Pull request by rgsl888prabhu**
Should have used base_size while creating the column from column view.

### CODE REVIEW

File: python/cudf/cudf/_libxx/copying.pyx [hunk 1]
```diff
@@ -243,12 +243,18 @@ def scatter(object input, object scatter_map, Table target,
         return _scatter_scalar(input, scatter_map, target, bounds_check)
 
 
-def scatter_to_tables(Table source_table, Column partition_map):
+def scatter_to_tables(Table source_table, Column partition_map,
+                      bool keep_index):
```
Comment (kkraus14) [inline python/cudf/cudf/_libxx/copying.pyx]:
```suggestion
                      bool keep_index=True):
```

File: python/cudf/cudf/core/frame.py [hunk 2]
```diff
@@ -352,14 +352,16 @@ def where(self, boolean_mask, other):
             result.index = self.index
             return result
 
-    def _scatter_to_tables(self, scatter_map):
+    def _scatter_to_tables(self, scatter_map, keep_index):
```
Comment (kkraus14) [inline python/cudf/cudf/core/frame.py]:
```suggestion
    def _scatter_to_tables(self, scatter_map, keep_index=True):
```

**Comment by rgsl888prabhu** [comment]
> Could we add a test case that verifies the fix?

@beckernick  As of now for that, we might have to add the parquet file or we can point it to remote folder ?

**Comment by rjzamora** [comment]
Thanks for the quick work on this @rgsl888prabhu !

**Pull request merged**